Hybrid/README.md

```diff
@@ -0,0 +1,131 @@
+---
+services: azure-resource-manager
+platforms: ruby
+author: devigned
+---
+
+[![Build Status](https://travis-ci.org/Azure-Samples/resource-manager-ruby-template-deployment.svg?branch=master)](https://travis-ci.org/Azure-Samples/resource-manager-ruby-template-deployment)
+
+# Deploy an SSH Enabled VM with a Template in Ruby
+
+This sample explains how to use Azure Resource Manager templates to deploy your Resources to Azure. It shows how to
+deploy your Resources by using the Azure SDK for Ruby.
+
+When deploying an application definition with a template, you can provide parameter values to customize how the
+resources are created. You specify values for these parameters either inline or in a parameter file.
+
+## Incremental and complete deployments
+
+By default, Resource Manager handles deployments as incremental updates to the resource group. With incremental
+deployment, Resource Manager:
+
+- leaves unchanged resources that exist in the resource group but are not specified in the template
+- adds resources that are specified in the template but do not exist in the resource group
+- does not re-provision resources that exist in the resource group in the same condition defined in the template
+
+With complete deployment, Resource Manager:
+
+- deletes resources that exist in the resource group but are not specified in the template
+- adds resources that are specified in the template but do not exist in the resource group
+- does not re-provision resources that exist in the resource group in the same condition defined in the template
+
+You specify the type of deployment through the Mode property, as shown in the examples below.
+
+## Deploy with Ruby
+
+In this sample, we are going to deploy a resource template which contains an Ubuntu 16.04 LTS virtual machine using
+ssh public key authentication, storage account, and virtual network with public IP address. The virtual network
+contains a single subnet with a single network security group rule which allows traffic on port 22 for ssh with a single
+network interface belonging to the subnet. The virtual machine is a `Standard_A1` size. You can find the template
+[here][Template].
+
+### To run this sample, do the following:
+
+You will need to create an Azure service principal either through Azure CLI, PowerShell or the portal. You should gather
+each the Tenant Id, Client Id and Client Secret from creating the Service Principal for use below.
+
+- [Create a Service Principal][ServicePrincipalCreation]
```

Comment:
Is there a sample on how to create a service principal through Ruby SDK? I have seen PS/Portal/CLI examples..

Reply:
I dont think so